src/doc/reference.md

```diff
@@ -3355,11 +3387,12 @@ more of the closure traits:
 ### Trait objects
 
 Every trait item (see [traits](#traits)) defines a type with the same name as
-the trait. This type is called the _trait object_ of the trait. Trait objects
-permit "late binding" of methods, dispatched using _virtual method tables_
-("vtables"). Whereas most calls to trait methods are "early bound" (statically
-resolved) to specific implementations at compile time, a call to a method on an
-trait objects is only resolved to a vtable entry at compile time. The actual
+the trait. Traits that are object-safe (see [RFC 255][rfc255]) can be
+instantiated as _trait objects_, which permit "late binding" of methods,
+dispatched using _virtual method tables_ ("vtables"). Whereas most calls to
+trait methods are "early bound" (statically resolved) to specific
+implementations at compile time, a call to a method on an trait objects is
+only resolved to a vtable entry at compile time. The actual
```

Comment:
This sentence threw me for a bit of a spin, I think the grammar around "on an trait objects is" may need to be updated slightly, but it was a little odd talking about late bindings, then early bindings, then back to late bindings. Perhaps this could be restructured ab it?